documentation/src/docs/retention.md

```diff
@@ -0,0 +1,158 @@
+<!--
+Copyright Pravega Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+# Pravega Stream Retention
+
+## Motivation
+Retention is an integral part of any data system. It is important to enable the application to remove their data for at least three reasons:
+   - To reclaim storage space
+   - To comply with legislation
+   - To comply with user requests
+
+We envision in Pravega two broad ways to implement retention:
+- Manual: The client API provides a call to truncate the data of a Stream at a given point. This point is given by an abstraction that we call a Stream Cut. A Stream Cut instance is an opaque object that internally contains a mapping of segments to offsets.
+- Automated: A Stream can be configured to be truncated automatically based on either time or size.
+
+## How Retention is handled by Pravega
+The Controller is responsible for managing retention of data in a Stream. 
+It does this by periodically truncating a Stream at a specific Stream Cut. 
+The choice of a Stream Cut depends on the retention policy set for the Stream.
+
+### Retention Policy - Size & Time Based
+
+The retention policy for a Pravega Stream specifies whether truncation happens based on size or time. 
+It contains a minimum limit (min limit) and a maximum limit (max limit) such that the Controller truncates data respecting those limits.
+For example, a size-based policy specifies the amount of data to retain, so if a retention policy specifies a minimum limit = 20 GB and a maximum limit = 100 GB, then the truncation cycle ensures that the Stream has at least 20 GB and not more than 100 GB.
+
+### Retention Set
+The Controller on every truncation cycle generates a Stream Cut SC at the tail end of the Stream and stores SC in a set called the retention set with the Stream metadata. A Stream Cut SC from the retention set is chosen to truncate the Stream.
```

Comment:
Parenthesize first instance of "SC", then consistently use abbreviation
-- "generates a Stream Cut (SC) at the tail"
&& "An SC from the retention set is chosen"